signalclient.go

```diff
@@ -147,7 +158,7 @@ func (c *SignalClient) connect(urlPrefix string, token string, params SignalClie
 	}
 
 	header := newHeaderWithToken(token)
-	conn, hresp, err := websocket.DefaultDialer.Dial(u.String(), header)
+	conn, hresp, err := websocket.DefaultDialer.DialContext(ctx, u.String(), header)
 	if err != nil {
```

Comment:
if the err is due to context cancelled, then we should skip checking `/validate`.

Reply:
ah just read the above comment stream. feel free to ignore :)